Exploration Queries/SecurityEvent/ProcessesOnHost.txt

```diff
@@ -1,35 +1,31 @@
-| where toupper(Computer) contains v_Host or toupper(WorkstationName) contains v_Host
 | where EventID == 4688
 // excluding well known processes, feel free to add more specific to the environment
-| where NewProcessName !contains ":\\Windows\\System32\\conhost.exe" and ParentProcessName !contains ":\\Windows\\System32\\conhost.exe"
-and NewProcessName !contains ":\\Windows\\Microsoft.NET\\Framework64\\v2.0.50727\\csc.exe" and ParentProcessName !contains ":\\Windows\\Microsoft.NET\\Framework64\\v2.0.50727\\csc.exe"
-and NewProcessName !contains ":\\Windows\\Microsoft.NET\\Framework64\\v2.0.50727\\cvtres.exe" and ParentProcessName !contains ":\\Windows\\Microsoft.NET\\Framework64\\v2.0.50727\\cvtres.exe"
-and NewProcessName!contains ":\\Program Files\\Microsoft Monitoring Agent\\Agent\\MonitoringHost.exe" and ParentProcessName !contains ":\\Program Files\\Microsoft Monitoring Agent\\Agent\\MonitoringHost.exe"
-and ParentProcessName !contains ":\\Windows\\CCM\\CcmExec.exe"
+| where NewProcessName !contains ':\\Windows\\System32\\conhost.exe' and ParentProcessName !contains ':\\Windows\\System32\\conhost.exe'
+and NewProcessName !contains ':\\Windows\\Microsoft.NET\\Framework64\\v2.0.50727\\csc.exe' and ParentProcessName !contains ':\\Windows\\Microsoft.NET\\Framework64\\v2.0.50727\\csc.exe'
+and NewProcessName !contains ':\\Windows\\Microsoft.NET\\Framework64\\v2.0.50727\\cvtres.exe' and ParentProcessName !contains ':\\Windows\\Microsoft.NET\\Framework64\\v2.0.50727\\cvtres.exe'
+and NewProcessName!contains ':\\Program Files\\Microsoft Monitoring Agent\\Agent\\MonitoringHost.exe' and ParentProcessName !contains ':\\Program Files\\Microsoft Monitoring Agent\\Agent\\MonitoringHost.exe'
+and ParentProcessName !contains ':\\Windows\\CCM\\CcmExec.exe'
 // excluding well known parent process, child process combinations, feel free to add more specific to the environment
-| where (ParentProcessName !contains ":\\Windows\\System32\\svchost.exe" and (NewProcessName !contains ":\\Windows\\System32\\wbem\\WmiPrvSE.exe" or NewProcessName !contains ":\\Windows\\SysWOW64\\wbem\\WmiPrvSE.exe"))
-| where (ParentProcessName !contains ":\\Windows\\System32\\services.exe" and NewProcessName !contains ":\\Windows\\servicing\\TrustedInstaller.exe")
-| summarize min(TimeGenerated), max(TimeGenerated) by Computer, Account, NewProcessName, CommandLine
-| project min_TimeGenerated, max_TimeGenerated, Computer, Account, NewProcessName, CommandLine
+| where (ParentProcessName !contains ':\\Windows\\System32\\svchost.exe' and (NewProcessName !contains ':\\Windows\\System32\\wbem\\WmiPrvSE.exe' or NewProcessName !contains ':\\Windows\\SysWOW64\\wbem\\WmiPrvSE.exe'))
+| where (ParentProcessName !contains ':\\Windows\\System32\\services.exe' and NewProcessName !contains ':\\Windows\\servicing\\TrustedInstaller.exe')
+| where toupper(Computer) contains v_Host_HostName or toupper(WorkstationName) contains v_Host_HostName                            
```

Comment:
I agree normally, but I am not sure it is more expensive here in this scenario. The query now is looking at all the events from all the computers in the table to exclude these NewProcessName using contains, whereas before it was matching the HostName first and then only iterating over the NewProcessName. In general, I think HostName is probably a smaller index, but maybe not.

Reply:
true